rest_tools/server/arghandler.py

```diff
@@ -0,0 +1,153 @@
+"""Handle argument parsing, defaulting, and casting."""
+
+
+import json
+from typing import Any, cast, Dict, List, Optional
+
+import tornado.web
+
+# local imports
+from rest_tools.client.json_util import json_decode
+
+
+class _NoDefaultValue:  # pylint: disable=R0903
+    """Signal no default value, AKA argument is required."""
+
+
+NO_DEFAULT = _NoDefaultValue()
+_FALSES = ["FALSE", "False", "false", "0", 0, "NO", "No", "no"]
+
+
+def _get_json_body(request_handler: tornado.web.RequestHandler) -> Dict[str, Any]:
+    json_body = json_decode(request_handler.request.body)  # type: ignore[no-untyped-call]
+    return cast(Dict[str, Any], json_body)
+
+
+class _UnqualifiedArgumentError(Exception):
+    """Raise when ArgumentHandler._qualify_argument() fails."""
+
+
+def _make_400_error(arg_name: str, error: Exception) -> tornado.web.HTTPError:
+    if isinstance(error, tornado.web.MissingArgumentError):
+        return error  # MissingArgumentError is already a 400 error
+    return tornado.web.HTTPError(400, reason=f"`{arg_name}`: {error}")
+
+
+class ArgumentHandler:
+    """Helper class for argument parsing, defaulting, and casting.
+
+    Like argparse.ArgumentParser, but for REST & JSON-body arguments.
+    """
+
+    @staticmethod
+    def _qualify_argument(
+        type_: Optional[type], choices: Optional[List[Any]], value: Any
+    ) -> Any:
+        """Cast `value` to `type_` type, and/or check `value` in in `choices`.
+
+        Raise _UnqualifiedArgumentError if either qualification fails.
+        """
+        if type_:
+            try:
+                if (type_ == bool) and (value in _FALSES):
```

Comment:
I'd rather compare to true values, since if I used "fAlse" that would be true in the current schema.  My default for an unknown value is false, at least.

But to at least fix the comparison,  use `.lower()` on any string type so case isn't a consideration.

Reply:
I just discovered `distutils.util.strtobool()`. That'll solve this problem

Reply:
^ the default is a `ValueError` in this case